src/mqtt_client.rs

```diff
@@ -416,6 +443,12 @@ where
                     };
                 }
 
+                // Replay QoS 1 messages
+                let to_send = self.session_state.pending_publish.clone();
```

Comment:
We definitely don't want to clone the entire map of pending messages. Instead, we should just iterate over the keys of the map and borrow the data out of it.

Reply:
Cloning keys and iterating over those.